app/modules/intelligence/agents/chat_agents/lld_chat_agent.py

```
@@ -68,21 +96,39 @@ async def _create_chain(self) -> RunnableSequence:
+                template=prompt,
+                partial_variables={"format_instructions": parser.get_format_instructions()},
+            )
+            chain = prompt_with_parser | self.llm | parser
+            response = await chain.ainvoke(input=inputs)
+
+            return response.classification
 
-        return response.classification
+        except asyncio.TimeoutError:
+            logger.error("Timeout waiting for rate limiter")
+            raise Exception("Service is currently overloaded. Please try again later.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)